cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -471,18 +471,18 @@ private module Stage1 {
   pragma[nomagic]
   private predicate revFlowConsCand(Content c, Configuration config) {
     exists(Node mid, Node node |
-      fwdFlow(node, unbind(config)) and
+      fwdFlow(node, pragma[only_bind_into](config)) and
       read(node, c, mid) and
-      fwdFlowConsCand(c, unbind(config)) and
-      revFlow(mid, _, config)
+      fwdFlowConsCand(c, pragma[only_bind_into](config)) and
+      revFlow(pragma[only_bind_into](mid), _, pragma[only_bind_into](config))
```

Comment:
Why is the pragma needed for `mid`?

Reply:
Because without it we may start the pipeline with
```
fwdFlowConsCand(c, pragma[only_bind_into](config)) and
not revFlowConsCand#prev(Content c, Configuration config)
```
which isn't loop-invariant and doesn't include the delta, thus making the predicate quadratic.